tests/integration/commands/init_command_test.py

```diff
@@ -0,0 +1,43 @@
+from pathlib import Path
+from filecmp import dircmp, cmpfiles
+
+import pytest
+
+from protostar.cairo import CairoVersion
+from tests.integration.conftest import CreateProtostarProjectFixture
+from tests.conftest import PROJECT_ROOT
+
+
+@pytest.mark.parametrize(
+    "create_protostar_project, cairo_version",
+    [
+        (CairoVersion.cairo0, CairoVersion.cairo0),
+        (CairoVersion.cairo1, CairoVersion.cairo1),
+    ],
+    indirect=["create_protostar_project"],
+)
+def test_init(
+    create_protostar_project: CreateProtostarProjectFixture, cairo_version: CairoVersion
+):
+    def compare_directories(dcmp: dircmp):
+        if dcmp.left_only or dcmp.right_only or dcmp.diff_files:
+            return False
+
+        _, diff, errors = cmpfiles(
+            dcmp.left, dcmp.right, dcmp.same_files, shallow=False
+        )
+        if diff or errors:
+            return False
+
+        for sub_dcmp in dcmp.subdirs.values():
+            if not compare_directories(sub_dcmp):
+                return False
```

Comment:
So this function goes only 1 level deep, right? Wouldn't it be better to go as deep as necessary? Maybe we should do a recursion here? Or maybe `subdirs` contains **all** the subdirectories no matter how deep they are? Quick look at https://docs.python.org/3/library/filecmp.html#filecmp.dircmp.subdirs didn't help :/

Reply:
But there is a recursive call inside the `if` statement... Or am I missing sth?

Reply:
ok, right `compare_directories` is recursive.